src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -2,75 +2,97 @@
 {
-    private ProductDao productDao;
+    @InjectMocks
+    private final ArrayListProductDao productDao = new ArrayListProductDao();
+
+    private List<Product> testProducts;
+
+    @Mock
+    private Product product1 = new Product();
+
+    @Mock
+    private Product product2 = new Product();
+
+    @Mock
+    private Product product3 = new Product();
+
+    @Mock
+    private Product productToSave = new Product();
 
     @Before
     public void setup() {
-        productDao = new ArrayListProductDao();
+        testProducts = new ArrayList<>();
+        Currency usd = Currency.getInstance("USD");
+        product1.setId(15L);
+        testProducts.add(product1);
+
+        product2.setPrice(new BigDecimal(130));
+        product2.setCurrency(usd);
+        product2.setStock(30);
+        testProducts.add(product2);
+
+        product3.setStock(0);
+        testProducts.add(product3);
+
+        productDao.setProductList(testProducts);
+
     }
 
     @Test
     public void testGetProduct() throws ProductNotFoundException {
-        Currency usd = Currency.getInstance("USD");
-        Product expectedProduct = new Product(3L, "sgs3", "Samsung Galaxy S III", new BigDecimal(300), usd, 5, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S%20III.jpg");
-        Product actualProduct = productDao.getProduct(3L);
-        assertEquals(expectedProduct, actualProduct);
+        Product actualProduct = productDao.getProduct(15L);
+        assertEquals(product1, actualProduct);
+
     }
 
     @Test(expected = ProductNotFoundException.class)
     public void testGetProductException() throws ProductNotFoundException {
-        Product actualProduct = productDao.getProduct(14L);
+        productDao.getProduct(14L);
+
     }
 
     @Test
-    public void testFindProductsWithNotNullPriceAndStockMoreThan0()
-    {
-        List<Product> products = productDao.findProducts();
-        int amountFalse = 0;
-        for (Product product: products) {
-            if (product.getPrice() == null || product.getStock() <= 0)
-                amountFalse++;
-        }
-        assertTrue(amountFalse == 0);
+    public void testFindProductsWithNotNullPriceAndStockMoreThan0() {
+        long amountFalse = productDao.findProducts()
+                .stream()
+                .filter(product -> product.getPrice() == null || product.getStock() <= 0)
+                .count();
+        assertEquals(0, amountFalse);
+
     }
 
     @Test
     public void testSaveNewOrExistingProduct() throws ProductNotFoundException {
-        Currency usd = Currency.getInstance("USD");
-        Product productToSave = new Product( "sgs_new", "New Product Test", new BigDecimal(300), usd, 5, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S%20III.jpg");
         productDao.save(productToSave);
         assertTrue(productToSave.getId() > 0);
```

Comment:
This will not work on mock object, you'll have to use verify(productToSave).setId() instead